modules/loader/src/main/scala/com/snowplowanalytics/snowplow/postgres/streaming/source.scala

```diff
@@ -83,36 +88,46 @@ object source {
         val pubsubConfig = PubsubGoogleConsumerConfig[F](onFailedTerminate = pubsubOnFailedTerminate[F])
         PubsubGoogleConsumer
           .subscribeAndAck[F, Either[BadData, Data]](blocker, project, subscription, pubsubErrorHandler[F], pubsubConfig)
+      case Source.LocalFs(path) =>
+        val store: Store[F] = FileStore[F](NioPaths.get("/"), blocker)
+        store.list(path, recursive = true)
+          .map { p =>
+            store.get(p, chunkSize = 32 * 1024)
+              .through(fs2.text.utf8Decode)
+              .through(fs2.text.lines)
+              .filter(_.nonEmpty)
+              .map(parseItem(purpose, _))
+          }.flatten
     }
 
   /**
-    * Parse Kinesis record into a valid Loader's record, either enriched event or self-describing JSON,
-    * depending on purpose of the Loader
+    * Parse Kinesis record into a valid Loader's record
     */
-  def parseRecord(kind: Purpose, record: CommittableRecord): Either[BadData, Data] = {
-    val string =
-      try StandardCharsets.UTF_8.decode(record.record.data()).toString.asRight[BadData]
-      catch {
-        case _: IllegalArgumentException =>
-          val payload = StandardCharsets.UTF_8.decode(Base64.getEncoder.encode(record.record.data())).toString
-          kind match {
-            case Purpose.Enriched =>
-              val badRow = BadRow.LoaderParsingError(Cli.processor, NotTSV, Payload.RawPayload(payload))
-              BadData.BadEnriched(badRow).asLeft
-            case Purpose.SelfDescribing =>
-              BadData.BadJson(payload, "Cannot deserialize self-describing JSON from Kinesis record").asLeft
-          }
+  def parseRecord(kind: Purpose, record: CommittableRecord): Either[BadData, Data] =
+    Either.fromTry(Try(StandardCharsets.UTF_8.decode(record.record.data()).toString))
```

Comment:
You can use `Either.catchNonFatal` (comes from `cats.syntax.either`) or `Either.catchOnly[IllegalArgumentException]`